products.go

```diff
@@ -1,22 +1,38 @@
 package tesla
 
+import (
+	"encoding/json"
+)
+
 type Product struct {
-	EnergySiteId      int64   `json:"energy_site_id,omitempty"`
-	ResourceType      string  `json:"resource_type"`
-	ID                string  `json:"id"`
-	AssetSiteId       string  `json:"asset_site_id,omitempty"`
-	GatewayId         string  `json:"gateway_id,omitempty"`
-	WarpSiteNumber    string  `json:"warp_site_number,omitempty"`
-	EnergyLeft        float64 `json:"energy_left,omitempty"`
-	TotalPackEnergy   uint64  `json:"total_pack_energy,omitempty"`
-	PercentageCharged float64 `json:"percentage_charged,omitempty"`
-	BatteryType       string  `json:"battery_type,omitempty"`
-	BackupCapable     bool    `json:"backup_capable,omitempty"`
-	BatteryPower      int64   `json:"battery_power,omitempty"`
+	EnergySiteId      int64      `json:"energy_site_id,omitempty"`
+	ResourceType      string     `json:"resource_type"`
+	ID                Identifier `json:"id"`
+	AssetSiteId       string     `json:"asset_site_id,omitempty"`
+	GatewayId         string     `json:"gateway_id,omitempty"`
+	WarpSiteNumber    string     `json:"warp_site_number,omitempty"`
+	EnergyLeft        float64    `json:"energy_left,omitempty"`
+	TotalPackEnergy   uint64     `json:"total_pack_energy,omitempty"`
+	PercentageCharged float64    `json:"percentage_charged,omitempty"`
+	BatteryType       string     `json:"battery_type,omitempty"`
+	BackupCapable     bool       `json:"backup_capable,omitempty"`
+	BatteryPower      int64      `json:"battery_power,omitempty"`
 
 	c *Client
 }
 
+type Identifier string
+
+func (i *Identifier) UnmarshalJSON(data []byte) error {
+	if n := len(data); n > 1 && data[0] == '"' && data[n-1] == '"' {
+		return json.Unmarshal(data, (*string)(i))
```

Comment:
Unless you want to guard against a potential sub structure thats probably not needed and you can cast to string right away?

Reply:
I was seeing potential for other control sequences (e.g. masked quotes), and felt this was the easiest way to get that taken care of.